src/features/home/view/first/home-first-search-box.components.tsx

```diff
@@ -0,0 +1,149 @@
+import { AutoComplete, Button, Form, Input, Rate, Spin } from "antd";
+import useTranslation from "next-translate/useTranslation";
+import { useRouter } from "next/router";
+import { useContext, useEffect, useState } from "react";
+import { AppContext } from "../../../../core/app/app.context";
+import { PagesUrls, TranslationFiles } from "../../../../core/core";
+import { SORT_IN_ELASTICSEARCH } from "../../../restaurants/constants/restaurants.constants";
+import { SearchResultsRestaurants } from "../../../restaurants/services/restaurants/models/search-results-restaurants.models";
+import { restaurantsServices } from "../../../restaurants/services/restaurants/restaurants.services";
+import useDebounce from "../../../shared/hooks/debounce.hooks";
+import classes from "./style.module.css";
+
+const HomeFirstSearchBox = () => {
+  const { direction } = useContext(AppContext);
+
+  const { t } = useTranslation(TranslationFiles.HOME);
+
+  const [form] = Form.useForm();
+
+  const [restaurantName, setRestaurantName] = useState("");
+
+  const [searchResultsForRestaurants, setSearchResultsForRestaurants] =
+    useState<SearchResultsRestaurants[]>([]);
+
+  const [options, setOptions] = useState<any>([]);
+
+  const debouncedSearchTerm = useDebounce(restaurantName, 500);
+
+  const { push } = useRouter();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (debouncedSearchTerm) {
+      getRestaurants(restaurantName);
+    }
+  }, [debouncedSearchTerm, restaurantName]);
```

Comment:
Here it's better for "useEffect" to be like this

  useEffect(() => {
    if (debouncedSearchTerm) {
      getRestaurants(debouncedSearchTerm);
    }
  }, [debouncedSearchTerm]);
  
  so we can benefit of debounce